packages/core/src/interpreter.ts

```diff
@@ -697,6 +697,11 @@ export class Actor<TLogic extends AnyActorLogic>
    * @see {@link Actor.getPersistedSnapshot} to persist the internal state of an actor (which is more than just a snapshot).
    */
   public getSnapshot(): SnapshotFrom<TLogic> {
+    if (isDevelopment && !this._snapshot) {
+      throw new Error(
+        `Snapshot can't be read while the actor initializes itself`
```

Comment:
```suggestion
        `Snapshot cannot be read before actor is started.`
```

Reply:
The suggested change is not correct. This is perfectly valid:
```ts
const actorRef = createActor(machine)
actorRef.getSnapshot()
actorRef.start()
```

This isn't:
```ts
const machine = createMachine({
  context: ({ self }) => {
    self.getSnapshot() // we can't do this
    return {}
  }
})
createActor(machine)
```

Reply:
Oh okay, makes sense